src/mlpa/core/http_client.py

```diff
@@ -0,0 +1,19 @@
+import httpx
+
+from mlpa.core.config import env
+
+_client: httpx.AsyncClient | None = None
+
+
+def get_http_client() -> httpx.AsyncClient:
+    global _client
+    if _client is None:
+        _client = httpx.AsyncClient(timeout=env.UPSTREAM_TIMEOUT_SECONDS)
```

Comment:
imho maybe it is worth to exposing connection pool limits to the infra code
like

```python
httpx.AsyncClient(
            limits=httpx.Limits(
                max_keepalive_connections=...
                max_connections=...
                keepalive_expiry=...
            ),
            timeout=httpx.Timeout(
                connect=...
                read=env.UPSTREAM_TIMEOUT_SECONDS,
                write=..
                pool=...
            ),
        )
```